android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java

```diff
@@ -49,821 +49,546 @@ of this software and associated documentation files (the "Software"), to deal
+        OneSignal.getInAppMessages().setInAppMessageClickHandler(null);
+        OneSignal.getInAppMessages().setInAppMessageLifecycleHandler(null);
+        OneSignal.getNotifications().setNotificationClickHandler(null);
+        OneSignal.getNotifications().setNotificationWillShowInForegroundHandler(null);
+    }
+
+    private void sendEvent(String eventName, Object params) {
+        mReactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit(eventName, params);
+    }
+
+    public RNOneSignal(ReactApplicationContext reactContext) {
+        super(reactContext);
+        mReactApplicationContext = reactContext;
+        mReactContext = reactContext;
+        mReactContext.addLifecycleEventListener(this);
+        notificationReceivedEventCache = new HashMap<String, INotificationReceivedEvent>();
+    }
+
+
+    /** Native Module Overrides */
+    @Override
+    public String getName() {
+        return "OneSignal";
+    }
+
+    @Override
+    public void onHostDestroy() {
+        removeHandlers();
+        removeObservers();
+    }
+
+    @Override
+    public void onHostPause() {}
+
+    @Override
+    public void onHostResume() {}
+
+    @Override
+    public void onCatalystInstanceDestroy() {
+        removeHandlers();
+        removeObservers();
+    }
+
+    // OneSignal namespace methods
+    @ReactMethod
+    public void initialize(String appId) {
+        Context context = mReactApplicationContext.getCurrentActivity();
+
+        if (oneSignalInitDone) {
+            Log.e("OneSignal", "Already initialized the OneSignal React-Native SDK");
+            return;
+        }
+
+        oneSignalInitDone = true;
+
+        if (context == null) {
+            // in some cases, especially when react-native-navigation is installed,
+            // the activity can be null, so we can initialize with the context instead
+            context = mReactApplicationContext.getApplicationContext();
+        }
+
+        OneSignal.initWithContext(context, appId);
+    }
+
+    @ReactMethod
+    public void setPrivacyConsent(Boolean value) {
+        OneSignal.setPrivacyConsent(value);
+    }
+
+    @ReactMethod
+    public void getPrivacyConsent(Promise promise) {
+        promise.resolve(OneSignal.getPrivacyConsent());
+    }
+
+    @ReactMethod
+    public void setRequiresPrivacyConsent(Boolean required) {
+        OneSignal.setRequiresPrivacyConsent(required);
+    }
+
+    @ReactMethod
+    public void getRequiresPrivacyConsent(Promise promise) {
+        promise.resolve(OneSignal.getRequiresPrivacyConsent());
+    }
+
+
+    // OneSignal.Debug namespace methods
+    @ReactMethod
+    public void setLogLevel(int logLevel) {
+        OneSignal.getDebug().setLogLevel(LogLevel.fromInt(logLevel));
+    }
+
+    @ReactMethod
+    public void setAlertLevel(int logLevel) {
+        OneSignal.getDebug().setAlertLevel(LogLevel.fromInt(logLevel));
+    }
+
+
+    // OneSignal.InAppMessages namespace methods
+    @ReactMethod
```

Comment:
should this be a ReactMethod or something done internally after `OneSignal.initWithContext`?

Reply:
Not sure what you mean here, Brian

Reply:
I wasn't sure if `setInAppMessageClickHandler` is being called by react, or internally by this java code. It looks like the former, so all good